actors/evm/src/interpreter/system.rs

```diff
@@ -90,7 +93,8 @@ impl<'r, BS: Blockstore, RT: Runtime<BS>> System<'r, BS, RT> {
                 .context_code(ExitCode::USR_ILLEGAL_STATE, "state not in blockstore")?,
             nonce: state.nonce,
             saved_state_root: Some(state_root),
-            bytecode: Some(state.bytecode),
+            bytecode: if delegate.is_some() { delegate } else { Some(state.bytecode) },
```

Comment:
Hm. This could end up storing this new bytecode. We should handle delegatecall bytecode externally, if possible.

Reply:
yeah, good point.

Reply:
done.